server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1523,16 +1523,14 @@ public boolean isSegmentReplicationAllowed() {
             logger.warn("Shard is in primary mode and cannot perform segment replication as a replica.");
             return false;
         }
-        if (this.routingEntry().primary() && this.routingEntry().isRelocationTarget() == false) {
-            logger.warn("Shard is marked as primary but not relocating, so cannot perform segment replication");
-            return false;
+        if (this.routingEntry().primary()) {
+            logger.warn("Shard routing is marked primary thus cannot perform segment replication as replica");
```

Comment:
Shouldn't we return false here, if shard is primary?

Reply:
Yes, `false` should be returned. Thanks for catching this. Updated.